pkg/wireguard/agent/agent.go

```diff
@@ -364,3 +424,94 @@ func loadOrGeneratePrivKey(filePath string) (key wgtypes.Key, err error) {
 
 	return wgtypes.NewKey(bytes)
 }
+
+// OnIPIdentityCacheChange implements ipcache.IPIdentityMappingListener
+func (a *Agent) OnIPIdentityCacheChange(modType ipcache.CacheModification, ipnet net.IPNet, oldHostIP, newHostIP net.IP,
+	oldID *identity.NumericIdentity, newID identity.NumericIdentity, encryptKey uint8, k8sMeta *ipcache.K8sMetadata) {
```

Comment:
Just an idea, but as we currently encrypt only pod2pod, we could avoid unnecessary updates for identities which we don't care about (i.e. non-endpoint identities).

Reply:
I'm not sure what you are proposing. We are actually are ignoring updates for e.g. non-endpoint identities in the function body here, as we only consider entries where the `hostIP` (`oldHostIP`/`newHostIP`) is not `nil`. Meaning all updates with `hostIP=nil` are ignored by the body of `OnIPIdentityCacheChange`.

The current IPCache interface does not have a function to only subscribe to events where `hostIP` is set.

Reply:
Actually, I realize now that I should add a `oldHostIP!=nil` check to the `Delete` case. It only works by accident right now.

Reply:
Fixed. The code now should be much more explicit in the kind of events it handles.